src/sortStudents.ts

```diff
@@ -1,16 +1,54 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+function calculateAverageGrade(grades: number[]): number {
+  return grades.reduce((sum, grade) => sum + grade, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((studentA: Student, studentB: Student): number => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? studentA[sortBy].localeCompare(studentB[sortBy])
+          : studentA[sortBy].localeCompare(studentB[sortBy]);
+
+      case SortType.AverageGrade: {
+        const avgGradeA = calculateAverageGrade(studentA.grades);
+        const avgGradeB = calculateAverageGrade(studentB.grades);
 
+        return order === 'asc'
+          ? avgGradeA - avgGradeB
+          : avgGradeB - avgGradeA;
+      }
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'asc'
+          ? Number(studentA[sortBy]) - Number(studentB[sortBy])
+          : Number(studentB[sortBy]) - Number(studentA[sortBy]);
+      default:
+        return 0;
```

Comment:
Use default cases for throw error.
```suggestion
        throw new Error('Incorrect sort type');
```